Frends.FixedWidthFlatFile.Parse/Frends.FixedWidthFlatFile.Parse/FixedWidthFlatFile.cs

```diff
@@ -0,0 +1,199 @@
+    public static Result Parse([PropertyTab] Input input, [PropertyTab] Options options)
+    {
+        var inputRows = new List<string>();
+        var headers = new List<string>();
+        var outputData = new List<Dictionary<string, dynamic>>();
+
+        //== Read input data ==/
+        using (var reader = new StringReader(input.FlatFileContent))
+        {
+            inputRows = ReadLinesToList(reader);
+        }
+
+        //== Parse header row ==/
+        switch (input.HeaderRow)
+        {
+            case HeaderRowType.Delimited:
+                char headerDelimiter = Convert.ToChar(input.HeaderDelimiter);
+                headers = SplitToList(inputRows.First(), headerDelimiter);
+                inputRows.RemoveAt(0);
+                break;
+            case HeaderRowType.FixedWidth:
+                headers = SplitToList(inputRows.First(), input.ColumnSpecifications);
+                inputRows.RemoveAt(0);
+                break;
+        }
+        if (headers.Count > 0)
+        {
+            // add header values as name if not set in column specification
+            var index = 0;
+            foreach (var header in headers)
+            {
+                if (string.IsNullOrEmpty(input.ColumnSpecifications[index].Name))
+                    input.ColumnSpecifications[index].Name = header;
+                index++;
+            }
+        }
+
+        //== Parse data rows ==/
+
+        // Skip rows?
+        if (options.SkipRows)
+        {
+            if (options.SkipRowsFromTop > 0)
+            {
+                // skipping more rows that exist?
+                if (options.SkipRowsFromTop >= inputRows.Count)
+                    inputRows.Clear();
+                else
+                    inputRows = inputRows.Skip(options.SkipRowsFromTop).ToList();
+            }
+            if (options.SkipRowsFromBottom > 0)
+            {
+                // skipping more rows that exist?
+                if (options.SkipRowsFromBottom >= inputRows.Count)
+                    inputRows.Clear();
+                else
+                    inputRows = inputRows.Take((inputRows.Count - options.SkipRowsFromBottom)).ToList();
+            }
+        }
+        //== Process data rows ==/
+        foreach (var dataRow in inputRows)
+        {
+            outputData.Add(ParseDataRow(dataRow, input.ColumnSpecifications));
+        }
+
+        return new Result { Data = outputData };
+    }
+
+    internal static List<string> ReadLinesToList(StringReader reader)
+    {
+        var allLines = new List<string>();
+        string line;
+        while (null != (line = reader.ReadLine()))
+        {
+            //skip empty lines
+            if (!string.IsNullOrWhiteSpace(line))
+                allLines.Add(line);
+        }
+
+        return allLines;
+    }
+
+    internal static List<string> SplitToList(string row, char delimiter)
```

Comment:
Is it necessary to have helper method which only executes Split and ToList?

Reply:
It has two implementation and to keep the calling code consistent there are two overload for this method - one with just a delimiter and another is the an array of column specs. It would be otherwise kind of polluting the logic of the calling method. This way we have a more uniform style for parsing a row of data.